src/react/Router.tsx

```diff
@@ -91,199 +70,142 @@ function AttemptLogin({ sessionManager }: AttemptLoginProps): React.ReactElement
+function InterfaceEdit(): React.ReactElement {
   const { interfaceName, interfaceMajor } = useParams();
   return (
-    <InterfacePage
-      interfaceName={interfaceName}
-      interfaceMajor={parseInt(interfaceMajor, 10)}
-      {...props}
-    />
+    <InterfacePage interfaceName={interfaceName} interfaceMajor={parseInt(interfaceMajor, 10)} />
   );
 }
 
-function RegisterDevice(props: PageProps): React.ReactElement {
+function RegisterDevice(): React.ReactElement {
   const searchQuery = new URLSearchParams(useLocation().search);
   const deviceId = searchQuery.get('deviceId') || '';
 
-  return <RegisterDevicePage deviceId={deviceId} {...props} />;
+  return <RegisterDevicePage deviceId={deviceId} />;
 }
 
-function GroupDevicesSubPath(props: PageProps): React.ReactElement {
+function GroupDevicesSubPath(): React.ReactElement {
   const { groupName } = useParams();
   const decodedGroupName = decodeURIComponent(groupName);
 
-  return <GroupDevicesPage groupName={decodedGroupName} {...props} />;
+  return <GroupDevicesPage groupName={decodedGroupName} />;
 }
 
-function FlowDetails(props: PageProps): React.ReactElement {
+function FlowDetails(): React.ReactElement {
   const { flowName } = useParams();
 
-  return <FlowDetailsPage flowName={flowName} {...props} />;
+  return <FlowDetailsPage flowName={flowName} />;
 }
 
-function FlowConfiguration(props: PageProps): React.ReactElement {
+function FlowConfiguration(): React.ReactElement {
   const [searchParams] = useSearchParams();
   const pipelineId = searchParams.get('pipelineId') || '';
 
-  return <FlowConfigurationPage pipelineId={pipelineId} {...props} />;
+  return <FlowConfigurationPage pipelineId={pipelineId} />;
 }
 
-function PipelineSubPath(props: PageProps): React.ReactElement {
+function PipelineSubPath(): React.ReactElement {
   const { pipelineId } = useParams();
 
-  return <PipelineSourcePage pipelineId={pipelineId} {...props} />;
+  return <PipelineSourcePage pipelineId={pipelineId} />;
 }
 
-function DeviceStatusSubPath(props: PageProps): React.ReactElement {
+function DeviceStatusSubPath(): React.ReactElement {
   const { deviceId } = useParams();
 
-  return <DeviceStatusPage deviceId={deviceId} {...props} />;
+  return <DeviceStatusPage deviceId={deviceId} />;
 }
 
-function BlockSubPath(props: PageProps): React.ReactElement {
+function BlockSubPath(): React.ReactElement {
   const { blockId } = useParams();
 
-  return <BlockSourcePage blockId={blockId} {...props} />;
+  return <BlockSourcePage blockId={blockId} />;
 }
 
-function DeviceDataSubPath(props: PageProps): React.ReactElement {
+function DeviceDataSubPath(): React.ReactElement {
   const { deviceId, interfaceName } = useParams();
 
-  return <DeviceInterfaceValues deviceId={deviceId} interfaceName={interfaceName} {...props} />;
-}
-
-interface NoMatchProps {
-  fallback: (requesteURL: string) => void;
+  return <DeviceInterfaceValues deviceId={deviceId} interfaceName={interfaceName} />;
```

Comment:
since we're using hook inside pages, we can use `useParams` hooks there aswell, removing subpaths definitions from here